### PR TITLE
@auth support for datastore and add has auth flag

### PR DESCRIPTION
**Pull request by SwaySway**
#### Description of changes

- `@auth` on sync query (conflict resolution)
- updated unit tests
- add has auth flag for deny by default with `@model`s that don't have `@auth`
- pass admin ui userpool id down into the resolver if admin ui is enabled

#### Checklist

- [ ] PR description included
- [ ] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#tests)
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-graphql-auth-transformer/src/__tests__/multi-auth.test.ts [hunk 1]
```diff
@@ -6,16 +6,22 @@ import {
...
 } from '@aws-amplify/graphql-transformer-core';
 import { DocumentNode, ObjectTypeDefinitionNode, Kind, FieldDefinitionNode, parse, InputValueDefinitionNode } from 'graphql';
 
-const noAuthModeDefaultConfig: AppSyncAuthConfiguration = {
-  defaultAuthentication: {
-    authenticationType: undefined,
-  },
-  additionalAuthenticationProviders: [],
-};
+// const noAuthModeDefaultConfig: AppSyncAuthConfiguration = {
```
Comment (cjihrig) [inline packages/amplify-graphql-auth-transformer/src/__tests__/multi-auth.test.ts:12]: Commented code.

File: packages/amplify-graphql-auth-transformer/src/__tests__/multi-auth.test.ts [hunk 2]
```diff
@@ -83,8 +82,8 @@ const privateWithApiKeyAuthDirective = '@auth(rules: [{allow: private, provider:
 const publicAuthDirective = '@auth(rules: [{allow: public}])';
 const publicUserPoolsAuthDirective = '@auth(rules: [{allow: public, provider: userPools}])';
 const privateAndPublicDirective = '@auth(rules: [{allow: private}, {allow: public}])';
-const privateAndPrivateIAMDirective = '@auth(rules: [{allow: private}, {allow: private, provider: iam}])';
 const privateIAMDirective = '@auth(rules: [{allow: private, provider: iam}])';
+// const privateAndPrivateIAMDirective = '@auth(rules: [{allow: private}, {allow: private, provider: iam}])';
```
Comment (cjihrig) [inline packages/amplify-graphql-auth-transformer/src/__tests__/multi-auth.test.ts:86]: Here as well.

File: packages/amplify-graphql-auth-transformer/src/graphql-auth-transformer.ts [hunk 3]
```diff
@@ -261,6 +261,9 @@ Static group authorization should perform as expected.`,
           case QueryFieldType.LIST:
             this.protectListResolver(context, def, query.typeName, query.fieldName, acm);
             break;
+          case QueryFieldType.SYNC:
+            // protect sync query
```
Comment (cjihrig) [inline packages/amplify-graphql-auth-transformer/src/graphql-auth-transformer.ts:265]: Should this be calling `protectSyncResolver()`?

**Review by cjihrig — APPROVED** [review]
LGTM with a few comments.

File: packages/amplify-graphql-auth-transformer/src/graphql-auth-transformer.ts (hunk 3 above)
Comment (SwaySway) [reply packages/amplify-graphql-auth-transformer/src/graphql-auth-transformer.ts:265]: missed this line on the git add will add in the next commit

**Pull request merged**